tests/Feature/Api/PublicBookTest.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Tests\Feature\Api;
+
+use App\Models\Book;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\TestCase;
+
+class PublicBookTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /** @test */
+    public function can_get_list_of_public_books_with_keyword()
+    {
+        $publicBook1 = factory(Book::class)->create([
+            'report_visibility_code' => 'public',
+            'status_id' => 1,
+            'name' => 'Book 1',
+        ]);
+
+        $publicBook2 = factory(Book::class)->create([
+            'report_visibility_code' => 'public',
+            'status_id' => 0,
+            'name' => 'Book 2',
+        ]);
+
+        $nonPublicBook = factory(Book::class)->create([
+            'report_visibility_code' => 'internal',
+            'status_id' => 1,
+            'name' => 'Non-Public Book',
+        ]);
+
+        $this->getJson(route('api.public_books.index', [
+            'search' => 'Book 1',
+        ]));
+
+        $this->seeJson([
+            'name' => $publicBook1->name,
+        ]);
+
+        $this->dontSeeJson([
```

Comment:
Ini mestinya `$publicBook2->name` tampil kan ya? Karena visibility nya `public` walaupun non aktif
```suggestion
        $this->seeJson([
```